addons/crm_mail_plugin/tests/test_crm_mail_plugin.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+
+from odoo.addons.mail_plugin.tests.common import TestMailPluginControllerCommon
+
+
+class TestCrmMailPlugin(TestMailPluginControllerCommon):
+    def test_get_contact_data(self):
+        """Check that the leads section is not visible if the user has not access to crm.lead."""
+        partner = self.env["res.partner"].create({"name": "Test"})
+
+        data = {
+            "id": 0,
+            "jsonrpc": "2.0",
+            "method": "call",
+            "params": {"partner_id": partner.id},
+        }
+
+        with self.mock_auth_method():
+            result = self.url_open(
+                "/mail_plugin/partner/get",
+                data=json.dumps(data).encode(),
+                headers={"Content-Type": "application/json"},
+            ).json()["result"]
+
+        self.assertNotIn("leads", result,
+            msg="The user has no access to crm.lead, the leads section should not be visible")
+
+        self.user_test.groups_id |= self.env.ref("sales_team.group_sale_salesman_all_leads")
+
+        with self.mock_auth_method():
+            result = self.url_open(
+                "/mail_plugin/partner/get",
+                data=json.dumps(data).encode(),
+                headers={"Content-Type": "application/json"},
+            ).json()["result"]
+
+        self.assertIn("leads", result,
```

Comment:
Would be nice to create a lead and check that it's contained in the result, while we are at it.